Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +352,873 @@ public override bool IsServerConnected()
+                                busFlow.Activities.Move(bfActIndx, startGroupActsIndxInBf + groupIndx);
+                            }
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex);
+                        //failed to re order the activities to match the tc steps order, not worth breaking the import because of this
+                    }
+                }
+
+                //Add the BF variables (linked variables)
+                if (busVariables.Keys.Count > 0)
+                {
+                    foreach (KeyValuePair<string, string> var in busVariables)
+                    {
+                        //add as String param
+                        VariableString busVar = new VariableString();
+                        busVar.Name = var.Key;
+                        busVar.InitialStringValue = var.Value;
+                        busFlow.AddVariable(busVar);
+                    }
+                }
+
+                return busFlow;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to import QC test set and convert it into " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow), ex);
+                return null;
+            }
+        }
+
+        public bool ExportBusinessFlow(BusinessFlow businessFlow, QCTestSet mappedTestSet, string fatherId, ObservableList<ExternalItemFieldBase> testSetFields, ObservableList<ExternalItemFieldBase> testInstanceFields, ref string result)
+        {
+            int testSetId = 0;
+            // ObservableList<ActivitiesGroup> existingActivitiesGroups = new ObservableList<ActivitiesGroup>();
+
+            try
+            {
+                if (mappedTestSet == null) //##create new Test Set in QC
+                {
+                    testSetId = CreateNewTestSet(businessFlow, fatherId, testSetFields);
+                }
+                else //##update existing test set
+                {
+                    testSetId = UpdateExistingTestSet(businessFlow, mappedTestSet, fatherId, testSetFields);
+                }
+
+                businessFlow.ExternalID = testSetId.ToString();
+                return true;
+            }
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- " + ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export the " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow) + " to QC/ALM", ex);
+                return false;
+            }
+        }
+        private int CreateNewTestSet(BusinessFlow businessFlow, string fatherId, ObservableList<ExternalItemFieldBase> testSetFields)
+        {
+            // foreach (ExternalItemFieldBase field in testSetFields)
+            //{
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)